src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpWindowsProxy.cs

```diff
@@ -294,7 +283,10 @@ public Uri GetProxy(Uri uri)
                         // we can return the Proxy uri directly.
                         if (proxyInfo.ProxyBypass == IntPtr.Zero)
                         {
-                            return GetUriFromString(Marshal.PtrToStringUni(proxyInfo.Proxy));
+                            string proxyString = Marshal.PtrToStringUni(proxyInfo.Proxy);
+                            ParseProxyConfig(proxyString, out Uri insecureProxy, out Uri secureProxy);
```

Comment:
Oh, if there's a PAC file, all that ParseProxyConfig logic is invoked for every single request?  I now care a lot more about perf ;)

Reply:
I'll come up with a quick benchmark; I'm worried that WinHTTP executing a PAC script is going to drown out anything we do here.

Reply:
Thanks.

Reply:
Yea, there's not much opportunity here unfortunately. Turning parsing off completely nets a ~2us improvement.

|       Method | ParseProxy |     Mean |    Error |   StdDev |
|------------- |----------- |---------:|---------:|---------:|
| ResolveProxy |      False | 138.8 us | 2.373 us | 2.220 us |
| ResolveProxy |       True | 141.0 us | 1.499 us | 1.402 us |

Reply:
Ok, thanks for the data.